lib/srv/desktop/discovery.go

```diff
@@ -271,10 +272,18 @@ func (s *WindowsService) ldapEntryToWindowsDesktop(ctx context.Context, entry *l
 		return nil, trace.Wrap(err)
 	}
 
+	// ensure no '.' in name, because we use SNI to route to the right
+	// desktop, and our cert is valid for *.desktop.teleport.cluster.local
+	name := strings.ReplaceAll(hostname, ".", "-")
+
+	// append portion of the object GUID to ensure that desktops from
+	// different domains that happen to have the same hostname don't conflict
+	if guid := entry.GetRawAttributeValue(windows.AttrObjectGUID); len(guid) >= 4 {
+		name += "-" + hex.EncodeToString(guid[:4])
```

Comment:
Is there any guarantee that the first 4 bytes are different?

Reply:
The first 4 bytes are the `time_low` field, which will not contain static data (such as the version identifier).

https://datatracker.ietf.org/doc/html/rfc4122#section-4.1.2